android/app/src/main/java/com/now/naaga/presentation/beginadventure/BeginAdventureActivity.kt

```diff
@@ -99,15 +115,33 @@ class BeginAdventureActivity : AppCompatActivity(), AnalyticsDelegate by Default
             startActivity(Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS))
             Toast.makeText(this, GPS_TURN_ON_MESSAGE, Toast.LENGTH_SHORT).show()
         } else {
-            startActivity(OnAdventureActivity.getIntent(this))
+            startActivity(getIntentWithAdventureOrWithout())
+            finish()
+        }
+    }
+
+    private fun getIntentWithAdventureOrWithout(): Intent {
+        val existingAdventure = getParcelableAdventure()?.toDomain()
+
+        return if (existingAdventure == null) {
+            OnAdventureActivity.getIntent(this)
+        } else {
+            OnAdventureActivity.getIntentWithAdventure(this, existingAdventure)
         }
     }
 
     companion object {
         private const val GPS_TURN_ON_MESSAGE = "GPS 설정을 켜주세요"
+        private const val ADVENTURE = "ADVENTURE"
 
         fun getIntent(context: Context): Intent {
             return Intent(context, BeginAdventureActivity::class.java)
         }
+
+        fun getIntentWithAdventure(context: Context, adventure: Adventure): Intent {
+            return Intent(context, BeginAdventureActivity::class.java).apply {
+                putExtra(ADVENTURE, adventure.toUi())
+            }
+        }
```

Comment:
```suggestion

        fun getIntent(context: Context, adventure: Adventure?): Intent {
            return Intent(context, BeginAdventureActivity::class.java).apply {
                adventure?.let {
                     putExtra(ADVENTURE, adventure.toUi())
                 }
            }
        }
```
이런 식으로 수정해보는건 어때요?
getIntent 함수를 분리해 알려주는 것 보단 getIntent 하나의 함수를 만들고 그 안에서 처리해주는 게 좋을 것 같아요!
크롱이 두가지 방법을 비교해보고 원하는 방식으로 진행하시면 됩니다~!

Reply:
좋은 방법인 것 같아요! 
하지만 적용했을 때 장점과 단점이 모두 존재하는 것 같아서 좀 더 생각해보고 필요에 따라 수정해보도록 할게요!